src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -33,30 +33,40 @@ namespace mpl = multipass::logging;
 namespace
 {
 constexpr auto category = "sshfs mount";
-template <typename Callable>
-auto run_cmd(mp::SSHSession& session, std::string&& cmd, Callable&& error_handler)
+auto run_cmd(mp::SSHSession& session, std::string&& cmd)
 {
     auto ssh_process = session.exec(cmd);
     if (ssh_process.exit_code() != 0)
-        error_handler(ssh_process);
-    return ssh_process.read_std_output();
-}
+        throw std::runtime_error(ssh_process.read_std_error());
 
-auto run_cmd(mp::SSHSession& session, std::string&& cmd)
-{
-    auto error_handler = [](mp::SSHProcess& proc) { throw std::runtime_error(proc.read_std_error()); };
-    return run_cmd(session, std::forward<std::string>(cmd), error_handler);
+    return ssh_process.read_std_output();
 }
 
-void check_sshfs_exists(mp::SSHSession& session)
+mp::SshfsMount::SshfsPkgType get_sshfs_package_type(mp::SSHSession& session)
```

Comment:
Could we do away with this, and just try the location for `multipass-sshfs`, then if that fails, `which sshfs`?

Reply:
Ok, I'll see what I can do.

Reply:
I redid this into a new function that detects which `sshfs` to use and adds the command options based on the environment and the `libfuse` version.